plutus-ir/src/Language/PlutusIR/Transform/Rename.hs

```diff
@@ -53,16 +56,19 @@ Two problems arise:
    'PLC.ScopedRenameM' is for performing the renaming (the second stage).
 -}
 
-instance (PLC.HasUnique (tyname ann) PLC.TypeUnique, PLC.HasUnique (name ann) PLC.TermUnique) =>
-        PLC.Rename (Term tyname name ann) where
-    rename = PLC.runScopedRenameM . renameTermM
+type instance PLC.HasUniques (Term tyname name ann) = PLC.HasUniques (PLC.Term tyname name ann)
```

Comment:
I wonder if we could bundle a bunch of these naming-related things into a class with some associated types. There are quite a few now.

Reply:
Sorry, what do you mean?

`HasUniques` is just a type family that returns the constraints needed for running the renamer or equality checks over some thing. I tried making it an [alias type class](https://gist.github.com/Icelandjack/5afdaa32f41adf3204ef9025d9da2a70#description), it didn't work, `HasUniques` really has to be transparent.